### PR TITLE
[DML] Add convTranspose2d op for DML backend

### PR DESCRIPTION
**Pull request by miaobin**
Support convTranspose2d op for DML backend. @mingmingtasd @fujunwei PTAL, thanks!

### CODE REVIEW

File: src/webnn/native/dml/GraphDML.cpp [hunk 1]
```diff
@@ -247,6 +247,30 @@ namespace webnn::native { namespace dml {
             return newFilterDims;
         }
 
+        std::vector<UINT> transposeFilterDimensionsAsIohw(
+            wnn::ConvTranspose2dFilterOperandLayout filterLayout,
+            const std::vector<UINT>& filterDims) {
+            std::vector<UINT> newFilterDims(4);
+            switch (filterLayout) {
+                case wnn::ConvTranspose2dFilterOperandLayout::Hwoi:
```
Comment (mingmingtasd) [inline src/webnn/native/dml/GraphDML.cpp:255]: How about return dimensions in each case like: {filterDims[0],filterDims[3],filterDims[1],filterDims[2]} without declaring newFilterDims? Could you help also refactor other transpose**DimensionsAs** functions?

File: src/webnn/native/dml/GraphDML.cpp [hunk 2]
```diff
@@ -1719,7 +1767,170 @@ namespace webnn::native { namespace dml {
...
+
+            std::vector<UINT> newInputDims = inputDims, newFilterDims = filterDims, newInputStrides,
+                              newFilterStrides;
+
+            const ConvTranspose2dOptions* options = convTranspose2d->GetOptions();
+
+            DML_TENSOR_DESC inputTensorDesc = inputEdge->outputTensorDESC;
+            if (options->inputLayout == wnn::InputOperandLayout::Nhwc) {
+                newInputDims = transposeDimensions(NhwcToNchw, inputDims);
+
```
Comment (mingmingtasd) [inline src/webnn/native/dml/GraphDML.cpp:1789]: Removing blank line 1789 may be better?

File: src/webnn/native/dml/GraphDML.cpp [hunk 3]
```diff
@@ -1719,7 +1767,170 @@ namespace webnn::native { namespace dml {
...
+                biasTensorDescPtr = &newBiasTensorDesc;
+                inputEdges.push_back(biasEdge);
+            }
+
+            std::vector<UINT> outputDims(4);
+            if (options->outputSizes != nullptr) {
+                std::vector<UINT> outputSizes;
+                outputSizes.assign(options->outputSizes,
+                                   options->outputSizes + options->outputSizesCount);
+                if (options->inputLayout == wnn::InputOperandLayout::Nchw) {
```
Comment (mingmingtasd) [inline src/webnn/native/dml/GraphDML.cpp:1844]: Could we get shape here like line 1856? You know we have calculated the output shape of each op.

**Comment by mingmingtasd** [comment]
Please rebase, thanks! @miaobin

File: src/webnn/native/dml/GraphDML.cpp (hunk 3 above)
Comment (miaobin) [reply src/webnn/native/dml/GraphDML.cpp]: Unlike conv2d, the outputSizes here is a parameter, and we can only get it directly if it is not set.

**Comment by miaobin** [comment]
> Please rebase, thanks! @miaobin

Rebased the code. @mingmingtasd  PTAL, thanks!

**Review by mingmingtasd — APPROVED** [review]
LGTM, thanks! @miaobin

File: src/webnn/native/dml/GraphDML.cpp [hunk 4]
```diff
@@ -1671,7 +1718,162 @@ namespace webnn::native ::dml {
...
+        DAWN_ASSERT(inputsOperand.size() == 2 || inputsOperand.size() == 3);
+        DAWN_ASSERT(mGraphEdgesMap.find(inputsOperand[0].Get()) != mGraphEdgesMap.end());
+        DAWN_ASSERT(mGraphEdgesMap.find(inputsOperand[1].Get()) != mGraphEdgesMap.end());
+
+        auto inputEdge = mGraphEdgesMap[inputsOperand[0].Get()];
+        auto filterEdge = mGraphEdgesMap[inputsOperand[1].Get()];
+
+        auto inputDims = ConvertDimensions(inputsOperand[0].Get()->Shape());
+        auto filterDims = ConvertDimensions(inputsOperand[1].Get()->Shape());
+        // auto outputDims = ConvertDimensions(convTranspose2d->Outputs()[0].Get()->Shape());
```
Comment (mingmingtasd) [inline src/webnn/native/dml/GraphDML.cpp:1731]: A nit, remove this line. @miaobin

Comment (miaobin) [reply src/webnn/native/dml/GraphDML.cpp]: Done.

**Review by fujunwei — APPROVED** [review]
Merge it.

**Pull request merged**